### PR TITLE
CB-7729 Retrieve errors from Data Lake Database backup/restore failures

### PR DESCRIPTION
**Pull request by hreeve-cloudera**
Pulls the status reason for failed database backup/restore operations from the
salt stderr ouput, instead of from the command comment. The is achieved by
adding a new YAML file used by SaltErrorResolver, stderrcommands.yaml. When a
command listed in that file is run, the stderr output will be used as the
resolved error message. The getErrorResultSummary() method in RunnerInfo now
returns the summary in map form for simpler parsing in SaltErrorResolver.

Tested with unit tests and by testing in my local cloudbreak dev environment.

See detailed description in the commit message.

### CODE REVIEW

File: core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/cluster/datalake/dr/BackupRestoreStatusService.java [hunk 1]
```diff
@@ -61,7 +61,15 @@ public void restoreDatabaseFinished(long stackId) {
 
     public void handleDatabaseRestoreFailure(long stackId, String errorReason, DetailedStackStatus detailedStatus) {
         clusterService.updateClusterStatusByStackId(stackId, Status.RESTORE_FAILED, errorReason);
-        stackUpdater.updateStackStatus(stackId, detailedStatus);
+        stackUpdater.updateStackStatus(stackId, detailedStatus, extractSaltErrorIfAvailable(errorReason));
         flowMessageService.fireEventAndLog(stackId, Status.UPDATE_FAILED.name(), DATALAKE_DATABASE_RESTORE_FAILED, errorReason);
     }
+
+    private String extractSaltErrorIfAvailable(String errorReason) {
+        String errors = "Error(s): ";
```
Comment (lajosrodek) [inline core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/cluster/datalake/dr/BackupRestoreStatusService.java:69]: Perhaps worth extracting as a (private) constant. 😄

Comment (hreeve-cloudera) [reply core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/cluster/datalake/dr/BackupRestoreStatusService.java:69]: Curse you, you found the one thing where I was like, "Should I make this a constant? Eh, it's not a big deal." :laughing:

I'll fix it in the next patch, but waiting to see if there are any other comments first.

File: orchestrator-salt/src/main/java/com/sequenceiq/cloudbreak/orchestrator/salt/SaltErrorResolver.java [hunk 2]
```diff
@@ -49,12 +55,15 @@ public void init() {
             String file = FileReaderUtils.readFileFromClasspath("salt/errormessages.yaml");
             errorMessages = new Yaml().load(file);
             LOGGER.info("Error messages for salt: {}", errorMessages);
+            file = FileReaderUtils.readFileFromClasspath("salt/stderrcommands.yaml");
+            commandsWithStderrFailures = new Yaml().load(file);
+            LOGGER.info("Salt commands that will pull the failure from stderr: {}", commandsWithStderrFailures);
```
Comment (kkalvagadda1) [inline orchestrator-salt/src/main/java/com/sequenceiq/cloudbreak/orchestrator/salt/SaltErrorResolver.java:60]: Do we know when the error is available in salt/errormessages.yaml and salt/stderrcommands.yaml files?

Comment (hreeve-cloudera) [reply orchestrator-salt/src/main/java/com/sequenceiq/cloudbreak/orchestrator/salt/SaltErrorResolver.java:60]: I'm not sure I understand the question. In errormessages.yaml, the code is basically replacing any errors run by the listed processes with the error in the YAML file. In stderrcommands.yaml we're just listing some processes, and the error comes from the stderr output. If the stderr output doesn't exist, then it defaults to a message that just says, "Failed to execute: <process string>"

File: core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/cluster/datalake/dr/BackupRestoreStatusService.java (hunk 1 above)
Comment (lajosrodek) [reply core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/cluster/datalake/dr/BackupRestoreStatusService.java:69]: Sure, definitely not a burning issue. 😛

**Comment by hreeve-cloudera** [comment]
recheck

**Pull request merged**